.github/workflows/release-phase-1-internal.yml

```diff
@@ -134,13 +134,13 @@ jobs:
         run: |-
           $FilePath = '.github/workflows/release-phase-1.yml'
           $FileContents = Get-Content -Raw $FilePath
-          $FileContents = $FileContents -replace '(?<Yaml>major:\s+default: )\d+', '${Yaml}${{ inputs.major }}'
-          $FileContents = $FileContents -replace '(?<Yaml>minor:\s+default: )\d+', '${Yaml}${{ inputs.minor }}'
-          $FileContents = $FileContents -replace '(?<Yaml>patch:\s+default: )', '${Yaml}${{ steps.increment-patch.outputs.INCREMENTED_PATCH }}'
+          $FileContents = $FileContents -replace '(?<Yaml>major:\s+default: )''\d+''', '${Yaml}''${{ inputs.major }}'''
+          $FileContents = $FileContents -replace '(?<Yaml>minor:\s+default: )''\d+''', '${Yaml}''${{ inputs.minor }}'''
+          $FileContents = $FileContents -replace '(?<Yaml>patch:\s+default: )''\d+''', '${Yaml}''${{ steps.increment-patch.outputs.INCREMENTED_PATCH }}'''
           $FileContents = $FileContents -replace 'inputs\.major \|\| \d+', 'inputs.major || ${{ inputs.major }}'
           $FileContents = $FileContents -replace 'inputs\.minor \|\| \d+', 'inputs.minor || ${{ inputs.minor }}'
           $FileContents = $FileContents -replace 'inputs\.patch \|\| \d+', 'inputs.patch || ${{ steps.increment-patch.outputs.INCREMENTED_PATCH }}'
-          Set-Content -Path $FilePath -Value $FileContents
+          Set-Content -Path $FilePath -Value $FileContents.Substring(0, $FileContents.Length - 1)
```

Comment:
Isn't this just the substring from index 0 to end?
If so, why is Substring even required?
Is it just to force the type to string or something?

Reply:
I don't understand exactly why this works, but it is a known issue that others have discussed as well. I've added a comment explaining what this is doing. And to be sure that it is right, I ran a build with and without. You can check the PRs in the history if you're interested.